### DISCUSSION TITLE
Problems when concatenating trajectories.

### DISCUSSION THREAD

**Discussion by FezokaTurbo** [MDAnalysis/mdanalysis#4653] [Q&A]
I don't know if my trajectories are, in any way, corrupted, so i tried to work with then to understand how they are responding:

```
import MDAnalysis as mda

import warnings
warnings.filterwarnings('ignore')

psf = 'Protein B/step3_input.psf'
dcd = 'Protein B/step5_production.dcd'
dcd2 = 'Protein B/step5_production_continued.dcd'

u = mda.Universe(psf, dcd, dcd2)
print(len(u.trajectory))

u.trajectory[963]
print(u.trajectory.ts)

u.trajectory[964]
print(u.trajectory.ts)

u = mda.Universe(psf, dcd, dcd2)
allatoms = u.select_atoms('all')

allatoms.write('Protein B/Vignaangularis_all.dcd', frames = u.trajectory)

print('Done.')
```
When I try to select the 964th frame of the trajectory, although the trajectory has 1500 frames, a error appers:
[https://prnt.sc/zPBiKxNPhten](url)

However, when I try to read the Vignaangularis_all.dcd, the len(u.trajectory) the output is 946:

```
import MDAnalysis as mda
import warnings
warnings.filterwarnings('ignore')

psf = 'Protein B/step3_input.psf'
dcd = 'Protein B/Vignaangularis_all.dcd'

u = mda.Universe(psf, dcd)

print(len(u.trajectory))
```

Am I the only one who has this issue? Where should I start to look to solve this?
Also, there's another problem:
Why does the time change when I write another trajectory file?

When I read the trajectory as such:
```
u = mda.Universe(psf, dcd, dcd2)
for ts in u.trajectory:
     print(ts.time)
```
The output is in picoseconds(ps), however, when it's read as such:
```
u = mda.Universe(psf, 'Protein B/Vignaangularis_all.dcd')
for ts in u.trajectory:
     print(ts.time)
```
The output in is (ps/100). Why?

**Comment by jochuan** [comment]
I also had a similar problem when trying to join the trajectories in another program such as VMD, for some motive the time of the frames changes from what it was originally.

**Comment by orbeckst** [comment]
Looks like a corrupted file to me or your assumption

> although the trajectory has 1500 frames

is wrong.

> Why does the time change when I write another trajectory file?

This may depend on the code that you're using.
